raiden/network/proxies/token_network.py

```diff
@@ -302,101 +387,130 @@ def settlement_timeout_max(self) -> int:
         """ Returns the maximal settlement timeout for the token network. """
         return self.proxy.contract.functions.settlement_timeout_max().call()
 
-    def locked_amount_by_locksroot(
-            self,
-            participant1: typing.Address,
-            participant2: typing.Address,
-            locksroot: typing.Locksroot,
-    ) -> int:
-        """ Returns the locked amount for a specific participant's locksroot. """
-        data = self._call_and_check_result(
-            'getParticipantLockedAmount',
-            to_checksum_address(participant1),
-            to_checksum_address(participant2),
-            locksroot,
-        )
-        return data
-
     def settle_block_number(
             self,
             participant1: typing.Address,
             participant2: typing.Address,
-    ) -> typing.BlockNumber:
-        """ Returns the channel settle_block_number. """
-        channel_data = self.detail_channel(participant1, participant2)
-        return channel_data.get('settle_block_number')
+            channel_identifier: typing.ChannelID = None,
+    ) -> typing.Optional[typing.BlockNumber]:
```

Comment:
General question:
in `CONTRIBUTING.md` it was mentioned that if the object takes `None` as a default value then we don't type it as `Optional`. Does this mean we only type return values which could be `None` as `Optional`?

Reply:
This is for them being as arguments. Here it's a return type. Can't put `= None` here

Reply:
I think you misunderstood the question. So this function might return None... but we choose  to use optional here while we don't in func arguments where the passed value or default value could be None. Don't you see this as an inconsistency? We might want to define the use of Optional as a type for return values but not parameters in this case.

Reply:
No I don't see this as an incosistency.